extension/src/prototypes/ui-components/TimelineParameterItem.tsx

```diff
@@ -0,0 +1,276 @@
+    activeTimelineComponentIdAtom,
+  );
+
+  const isActive = useRef(false);
+  const [playState, setPlayState] = useState<"play" | "pause" | "reverse" | undefined>(undefined);
+
+  const [speed, setSpeed] = useState(speedOptions[0].value);
+
+  const handleSpeedChange = useCallback(
+    (event: SelectChangeEvent<number>) => {
+      setSpeed(Number(event.target.value));
+      if (playState === "play" || playState === "reverse") {
+        window.reearth?.clock?.setSpeed?.(
+          (playState === "reverse" ? -1 : 1) * Number(event.target.value),
+        );
+      }
+    },
+    [playState],
+  );
+
+  const handlePlay = useCallback(() => {
+    setActiveTimelineComponentId(id);
+    if (currentDate >= endDate) return;
+
+    setPlayState("play");
+    isActive.current = true;
+    window.reearth?.clock?.setTime?.({
+      start: startDate,
+      stop: endDate,
+      current: currentDate,
+    });
+    window.reearth?.clock?.setSpeed?.(speed);
+    window.reearth?.clock?.play?.();
+  }, [startDate, endDate, currentDate, speed, id, setActiveTimelineComponentId]);
+
+  const handlePause = useCallback(() => {
+    setActiveTimelineComponentId(id);
+    setPlayState("pause");
+    isActive.current = true;
+    window.reearth?.clock?.pause?.();
+  }, [id, setActiveTimelineComponentId]);
+
+  const handlePlayReverse = useCallback(() => {
+    setActiveTimelineComponentId(id);
+    if (currentDate <= startDate) return;
+
+    setPlayState("reverse");
+    isActive.current = true;
+    window.reearth?.clock?.setTime?.({
+      start: startDate,
+      stop: endDate,
+      current: currentDate,
+    });
+    window.reearth?.clock?.setSpeed?.(-speed);
+    window.reearth?.clock?.play?.();
+  }, [startDate, endDate, currentDate, speed, id, setActiveTimelineComponentId]);
+
+  const handleJumpTime = useCallback(
+    (_: Event, value: number | number[]) => {
+      if (typeof value === "number") {
+        setActiveTimelineComponentId(id);
+        setPlayState("pause");
+        isActive.current = true;
+        window.reearth?.clock?.pause?.();
+        window.reearth?.clock?.setTime?.({
+          start: startDate,
+          stop: endDate,
+          current: new Date(value),
+        });
+      }
+    },
+    [id, startDate, endDate, setActiveTimelineComponentId],
+  );
+
+  const stateRef = useRef({ endDate, startDate, playState, handlePause });
+  stateRef.current = { endDate, startDate, playState, handlePause };
+
+  const onTick = useCallback((current: Date) => {
+    if (isActive.current) {
+      setCurrentDate(current);
+      if (
+        (current >= stateRef.current.endDate && stateRef.current.playState === "play") ||
+        (current <= stateRef.current.startDate && stateRef.current.playState === "reverse")
+      ) {
+        stateRef.current.handlePause();
+      }
+    }
+  }, []);
+
+  useEffect(() => {
+    window.reearth?.clock?.setRangeType?.("clamped");
+    window.reearth?.on?.("tick", onTick);
```

Comment:
We should avoid invoking reearth API in ui-component. Because this should be reusable component. So could you change to use callback like `onTickXXX`.

Reply:
I've moved invoking reearth API part to `shared/reearth/hooks/useTimeline.ts`. Since `current` is relies on tick event so the ui component might not work well without the hooks. 
Please let me know if you think should do more refactor on this.